king_phisher/client/assistants/campaign.py

```diff
@@ -156,6 +278,61 @@ def _set_comboboxes(self):
 				combobox.add_attribute(renderer, 'text', 2)
 			combobox.set_model(rpc.get_tag_model(tag_table, model=model))
 
+	def _load_url_treeview_tsafe(self, hostname=None, refresh=False):
+		if refresh or not self._url_information['created']:
+			self._url_information['data'] = self.application.rpc.graphql_find_file('get_site_templates.graphql')
+			self._url_information['created'] = datetime.datetime.utcnow()
+		gui_utilities.glib_idle_add_once(self.gobjects['treeselection_url_selector'].unselect_all)
+		self._url_selected = None
+		gui_utilities.glib_idle_add_once(self._url_model.clear)
+		gui_utilities.glib_idle_add_once(self._domain_list_store.clear)
+
+		url_information = self._url_information['data']
+		if not url_information:
+			return
+
+		domains = []
+		self._url_classifiers = {}
+		for edge in url_information['siteTemplates']['edges']:
+			for page in edge['node']['metadata']['pages']:
+				if hostname:
+					if not edge['node']['hostname'].startswith(hostname):
+						continue
+				domains.append(edge['node']['hostname'])
+				self._url_classifiers[edge['node']['hostname']] = edge['node']['metadata']['classifiers']
```

Comment:
This logic seems buggy. The hostname is not enough to uniquely identify this set of classifiers because multiple templates could share the same hostname. You should use the `(edge['node']['hostname'], page)` tuple or the url as the key to ensure you're not overwriting classifiers from other templates.

You're also calculating the `_url_classifiers` when you load the raw data, so that one at least makes alot of sense to me.